code/datums/viewport.dm

```diff
@@ -0,0 +1,72 @@
+/client/ // Holder Variables
+	var/last_view_x_dim = 7
+	var/last_view_y_dim = 7
+
+// Used in OnResize() - Holds the clients prefs.
+GLOBAL_VAR_INIT(lock_client_view_x, null)
+GLOBAL_VAR_INIT(lock_client_view_y, null)
+GLOBAL_LIST_INIT(valid_icon_sizes, list(32, 48, 64, 96, 128))
+
+// We hold this so we don't hit prefs often.
+/datum/preferences/var/icon_size = 64
+
+/client/verb/SetWindowIconSize(val as num|text)
+	set hidden = 1
+	winset(src, "mapwindow.map", "icon-size=[val]")
+	if(val != prefs?.icon_size)
+		prefs.icon_size = val ? val : prefs.read_preference(/datum/preference/numeric/pixel_size)
+		prefs.save_preferences()
+	OnResize()
+
+/client/verb/OnResize()
```

Comment:
why is this a verb at all

Reply:
it might break as a proc, not sure if the byond ui can call it then

Reply:
ah I see. Fuck